src/graphql/language/visitor.py

```diff
@@ -197,10 +200,18 @@ def __init_subclass__(cls) -> None:
 
     def get_visit_fn(self, kind: str, is_leaving: bool = False) -> Callable:
         """Get the visit function for the given node kind and direction."""
+
+        key = (kind, is_leaving)
+        if key in self._visit_fns:
+            return self._visit_fns[key]
+
         method = "leave" if is_leaving else "enter"
         visit_fn = getattr(self, f"{method}_{kind}", None)
         if not visit_fn:
             visit_fn = getattr(self, method, None)
+
+        self._visit_fns[key] = visit_fn
+
         return visit_fn
```

Comment:
This can be done more cleanly:

```suggestion

        key = (kind, is_leaving)
        if key not in self._visit_fns:
            method = "leave" if is_leaving else "enter"
            fn = getattr(self, f"{method}_{kind}", None)
            self._visit_fns[key] = fn or getattr(self, method, None)
        return self._visit_fns[key]
```

Reply:
LGTM with the change proposed by Syrus.

A `try ... except KeyError` might be even faster depending how often the visitor is reused. You may want to to some benchmarking here.